MatrixSDK/Room/Polls/PollAggregator.swift

```diff
@@ -54,40 +58,65 @@ public class PollAggregator {
     public var delegate: PollAggregatorDelegate?
     
     deinit {
-        room.removeListener(eventListener)
+        room.removeListener(referenceEventsListener)
+        session.aggregations.removeListener(editEventsListener as Any)
     }
     
-    public init(session: MXSession, room: MXRoom, pollStartEvent: MXEvent) throws {
+    public init(session: MXSession, room: MXRoom, pollStartEventId: String) throws {
         self.session = session
         self.room = room
-        self.pollStartEvent = pollStartEvent
+        self.pollStartEventId = pollStartEventId
+        self.pollBuilder = PollBuilder()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(handleRoomDataFlush), name: NSNotification.Name.mxRoomDidFlushData, object: self.room)
         
-        guard let pollStartEventContent = MXEventContentPollStart(fromJSON: pollStartEvent.content) else {
+        editEventsListener = session.aggregations.listenToEditsUpdate(inRoom: self.room.roomId) { [weak self] event in
+            guard let self = self,
+                  self.pollStartEventId == event.relatesTo.eventId
+            else {
+                return
+            }
+            
+            do {
+                try self.buildPollStartContent()
+            } catch {
+                self.delegate?.pollAggregator(self, didFailWithError: PollAggregatorError.invalidPollStartEvent)
+            }
+        }
+        
+        try buildPollStartContent()
+    }
+    
+    private func buildPollStartContent() throws {
+        guard let event = self.session.store.event(withEventId: self.pollStartEventId, inRoom: room.roomId),
+              let pollStartEventContent = MXEventContentPollStart(fromJSON: event.content) else {
             throw PollAggregatorError.invalidPollStartEvent
         }
         
         self.pollStartEventContent = pollStartEventContent
         
-        pollBuilder = PollBuilder()
+        self.hasBeenEdited = (event.unsignedData.relations?.replace != nil)
         
-        poll = pollBuilder.build(pollStartEventContent: self.pollStartEventContent, events: self.events, currentUserIdentifier: session.myUserId)
+        self.poll = self.pollBuilder.build(pollStartEventContent: pollStartEventContent,
```

Comment:
I'll stop adding these everywhere 🙃
```suggestion
        poll = pollBuilder.build(pollStartEventContent: pollStartEventContent,
```